### PR TITLE
[AWS SageMaker] Add more unit tests

### PR DESCRIPTION
**Pull request by akartsky**
Added more tests for following components :
Deploy            (98% coverage)
Ground_truth  (98% coverage)
Workteam (96% coverage)
Model (97% coverage)
Batch Transform (98% coverage)

https://github.com/kubeflow/pipelines/issues/3782

```
====================================================================================== 94 passed, 3 warnings in 1.67s ======================================================================================
Name                                                      Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------------
/app/batch_transform/src/batch_transform.py                  51      1    98%   75
/app/common/_utils.py                                       567     31    95%   62, 73-83, 88-92, 693, 710, 714, 724, 727, 738-742, 804-805, 816-817, 821-824, 834, 870, 873-874
/app/deploy/src/deploy.py                                    45      1    98%   67
/app/ground_truth/src/ground_truth.py                        54      1    98%   76
/app/hyperparameter_tuning/src/hyperparameter_tuning.py      67      1    99%   93
/app/model/src/create_model.py                               32      1    97%   54
/app/train/src/train.py                                      52      1    98%   80
/app/workteam/src/workteam.py                                27      1    96%   49
---------------------------------------------------------------------------------------
TOTAL                                                       895     38    96%
```

These tests check : (for 1 inout)
1. Argument parsing is proper 
2. main() function is proper
3. emulate boto3 client and verify the output 

These things can be done more extensively to verify every combination of input

### CODE REVIEW

File: components/aws/sagemaker/batch_transform/src/batch_transform.py [hunk 1]
```diff
@@ -56,7 +57,7 @@ def create_parser():
 
 def main(argv=None):
   parser = create_parser()
-  args = parser.parse_args()
+  args = parser.parse_args(argv)
```
Comment (gautamkmr) [inline components/aws/sagemaker/batch_transform/src/batch_transform.py:60]: why do we need this change with line 77 ?

Comment (akartsky) [reply components/aws/sagemaker/batch_transform/src/batch_transform.py:60]: in main, parse_args() automatically takes argv input if we don't specify anything.
But when we mock the function it doesn’t.

So we have to specify the parameter

**Comment by RedbackThomson** [comment]
Could you please run these with the automated unit and integration test CodeBuild jobs to ensure they all still pass successfully? Otherwise I think I'm happy for this to be merged.

**Comment by akartsky** [comment]
> Could you please run these with the automated unit and integration test CodeBuild jobs to ensure they all still pass successfully? Otherwise I think I'm happy for this to be merged.

will do 👍🏻

**Pull request merged**